src/axom/primal/geometry/Hexahedron.hpp

```diff
@@ -0,0 +1,398 @@
+                          PointType::midpoint(m_points[4], m_points[5]));
+
+    PointType fm3 =
+      PointType::midpoint(PointType::midpoint(m_points[0], m_points[3]),
+                          PointType::midpoint(m_points[4], m_points[7]));
+
+    PointType fm4 =
+      PointType::midpoint(PointType::midpoint(m_points[1], m_points[2]),
+                          PointType::midpoint(m_points[5], m_points[6]));
+
+    PointType fm5 =
+      PointType::midpoint(PointType::midpoint(m_points[2], m_points[3]),
+                          PointType::midpoint(m_points[6], m_points[7]));
+
+    PointType fm6 =
+      PointType::midpoint(PointType::midpoint(m_points[4], m_points[5]),
+                          PointType::midpoint(m_points[6], m_points[7]));
+
+    // Initialize tets
+    tets[0] = TetrahedronType(hc, m_points[1], m_points[0], fm1);
+    tets[1] = TetrahedronType(hc, m_points[0], m_points[3], fm1);
+    tets[2] = TetrahedronType(hc, m_points[3], m_points[2], fm1);
+    tets[3] = TetrahedronType(hc, m_points[2], m_points[1], fm1);
+
+    tets[4] = TetrahedronType(hc, m_points[4], m_points[0], fm2);
+    tets[5] = TetrahedronType(hc, m_points[0], m_points[1], fm2);
+    tets[6] = TetrahedronType(hc, m_points[1], m_points[5], fm2);
+    tets[7] = TetrahedronType(hc, m_points[5], m_points[4], fm2);
+
+    tets[8] = TetrahedronType(hc, m_points[3], m_points[0], fm3);
+    tets[9] = TetrahedronType(hc, m_points[0], m_points[4], fm3);
+    tets[10] = TetrahedronType(hc, m_points[4], m_points[7], fm3);
+    tets[11] = TetrahedronType(hc, m_points[7], m_points[3], fm3);
+
+    tets[12] = TetrahedronType(hc, m_points[5], m_points[1], fm4);
+    tets[13] = TetrahedronType(hc, m_points[1], m_points[2], fm4);
+    tets[14] = TetrahedronType(hc, m_points[2], m_points[6], fm4);
+    tets[15] = TetrahedronType(hc, m_points[6], m_points[5], fm4);
+
+    tets[16] = TetrahedronType(hc, m_points[6], m_points[2], fm5);
+    tets[17] = TetrahedronType(hc, m_points[2], m_points[3], fm5);
+    tets[18] = TetrahedronType(hc, m_points[3], m_points[7], fm5);
+    tets[19] = TetrahedronType(hc, m_points[7], m_points[6], fm5);
+
+    tets[20] = TetrahedronType(hc, m_points[7], m_points[4], fm6);
+    tets[21] = TetrahedronType(hc, m_points[4], m_points[5], fm6);
+    tets[22] = TetrahedronType(hc, m_points[5], m_points[6], fm6);
+    tets[23] = TetrahedronType(hc, m_points[6], m_points[7], fm6);
+  }
+
+  /**
+   * \brief Finds the volume of the hexahedron by triangulating the hexahedron
+   *        into 24 tetrahedrons and finding the total volume of the tetrahedrons
+   *
+   * \return The volume of the hexahedron
+   */
+  AXOM_HOST_DEVICE
+  double volume_tet_decomp()
+  {
+    double retVol = 0.0;
+    TetrahedronType tets[NUM_TRIANGULATE];
+
+    triangulate(tets);
+
+    for(int i = 0; i < NUM_TRIANGULATE; i++)
+    {
+      retVol += tets[i].volume();
+    }
+
+    return retVol;
+  }
```

Comment:
Is this function only here for testing? or is there another reason users might want it?

If it's only for testing, I'd suggest removing it from the class.